### PR TITLE
Fuzzer updates

### PR DESCRIPTION
**Pull request by kroeckx**

### CODE REVIEW

File: fuzz/README.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ https://github.com/llvm-mirror/llvm/tree/master/lib/Fuzzer if you prefer):
     $ sudo apt-get install subversion
     $ mkdir svn-work
     $ cd svn-work
-    $ svn co http://llvm.org/svn/llvm-project/llvm/trunk/lib/Fuzzer
+    $ svn co https://llvm.org/svn/llvm-project/compiler-rt/trunk/lib/fuzzer Fuzzer
```
Comment (richsalz) [inline fuzz/README.md]: !!! :)

File: fuzz/asn1.c [hunk 2]
```diff
@@ -201,6 +203,75 @@ static ASN1_ITEM_EXP *item_type[] = {
 
 static ASN1_PCTX *pctx;
 
+#define DO_TEST(TYPE, D2I, I2D, PRINT) { \
+    const unsigned char *p = buf; \
+    unsigned char *der = NULL; \
+    \
```
Comment (richsalz) [inline fuzz/asn1.c]: blank line after the TYPE, not before.

File: fuzz/asn1.c [hunk 3]
```diff
@@ -201,6 +203,75 @@ static ASN1_ITEM_EXP *item_type[] = {
...
+        OPENSSL_free(der); \
+        \
+        TYPE ## _free(type); \
+    } \
+}
+
+#define DO_TEST_PRINT_OFFSET(TYPE, D2I, I2D, PRINT) { \
+    const unsigned char *p = buf; \
+    unsigned char *der = NULL; \
+    \
```
Comment (richsalz) [inline fuzz/asn1.c]: blank line after not before.

File: fuzz/asn1.c [hunk 4]
```diff
@@ -201,6 +203,75 @@ static ASN1_ITEM_EXP *item_type[] = {
...
+        OPENSSL_free(der); \
+        \
+        TYPE ## _free(type); \
+    } \
+}
+
+#define DO_TEST_PRINT_PCTX(TYPE, D2I, I2D, PRINT) { \
+    const unsigned char *p = buf; \
+    unsigned char *der = NULL; \
+    \
```
Comment (richsalz) [inline fuzz/asn1.c]: you know what i'm going to say :)

File: fuzz/asn1.c [hunk 5]
```diff
@@ -201,6 +203,75 @@ static ASN1_ITEM_EXP *item_type[] = {
...
+        \
+        TYPE ## _free(type); \
+    } \
+}
+
+
+#define DO_TEST_NO_PRINT(TYPE, D2I, I2D) { \
+    const unsigned char *p = buf; \
+    unsigned char *der = NULL; \
+    \
```
Comment (richsalz) [inline fuzz/asn1.c]: :)

**Comment by kroeckx** [comment]
You missed like most of them

**Comment by richsalz** [comment]
I stopping nit-picking after I think the point was clear. :)

File: fuzz/asn1.c [hunk 6]
```diff
@@ -231,16 +301,38 @@ int FuzzerTestOneInput(const uint8_t *buf, size_t len)
...
+    DO_TEST_NO_PRINT(ESS_CERT_ID, d2i_ESS_CERT_ID, i2d_ESS_CERT_ID);
+    DO_TEST_NO_PRINT(ESS_SIGNING_CERT, d2i_ESS_SIGNING_CERT, i2d_ESS_SIGNING_CERT);
+    DO_TEST(DH, d2i_DHparams, i2d_DHparams, DHparams_print);
+    DO_TEST(DH, d2i_DHxparams, i2d_DHxparams, DHparams_print);
+    DO_TEST_NO_PRINT(DSA_SIG, d2i_DSA_SIG, i2d_DSA_SIG);
+    DO_TEST_PRINT_OFFSET(DSA, d2i_DSAPrivateKey, i2d_DSAPrivateKey, DSA_print);
+    DO_TEST_PRINT_OFFSET(DSA, d2i_DSAPublicKey, i2d_DSAPublicKey, DSA_print);
+    DO_TEST(DSA, d2i_DSAparams, i2d_DSAparams, DSAparams_print);
+    DO_TEST_PRINT_OFFSET(RSA, d2i_RSAPublicKey, i2d_RSAPublicKey, RSA_print);
+    DO_TEST_PRINT_OFFSET(EC_GROUP, d2i_ECPKParameters, i2d_ECPKParameters, ECPKParameters_print);
```
Comment (kaduk) [inline fuzz/asn1.c]: The no-ec builds are now complaining.    It looks like no-dsa, no-ts, etc., might also cause problems.

Comment (kroeckx) [reply fuzz/asn1.c]: Care to open a pull request for that?

Comment (kaduk) [reply fuzz/asn1.c]: I figured I'd give you first dibs if you wanted.  #4664

**Pull request closed without merging**